### PR TITLE
Add ReScript support

### PR DESCRIPTION
**Pull request by sorenhoyer**
Working example: https://github.com/sorenhoyer/react-relay-rescript-boilerplate

Todos:

- [x] Investigate issue when using variables in a query https://github.com/zth/reason-relay/issues/114#issuecomment-711285829
- [x] Write tests

### CODE REVIEW

**Comment by zth** [comment]
Great stuff! Let me review/merge this in the coming days! 🙂

**Comment by zth** [comment]
@sorenhoyer let me know if you want to discuss adding tests etc.

**Comment by sorenhoyer** [comment]
@zth I'd like that. I Added one test for parsing of a query with a variable with ReScript syntax, but before I add more, I'd like to hear what you had in mind. :-)

**Comment by zth** [comment]
No that looks great! I think maybe extending them to making sure the can handle multiple tags in the same file, and also that what it picks up is valid GraphQL (so just parsing it via graphql-js without doing anything with the results) would be a good idea too, if you don't mind.

**Comment by sorenhoyer** [comment]
@zth added multiple tags to the test and also parsing it with the relay-compiler to verify that it is valid graphql that is picked up.

Not sure what else is necessary to test here, other than perhaps the default reason syntax, but maybe that is a little out of scope for this PR, since this is focussing on adding ReScript support.

Let me know if you have any suggestions regarding the tests. I'm not so "creative" when it comes to finding test cases heh. ;-)

**Review by zth — COMMENTED** [review]
Great work! 😄 Just one comment and this is good to go 👍

File: packages/reason-relay/language-plugin/src/FindGraphQLTags.ts [hunk 1]
```diff
@@ -3,12 +3,12 @@ import { GraphQLTag } from "relay-compiler/lib/language/RelayLanguagePluginInter
...
 
 function parseFile(text: string, file: string) {
-  if (!text.includes("[%relay.")) {
+  if (!text.includes("[%relay.") && !text.includes("%relay.")) {
     return [];
   }
 
   invariant(
-    text.indexOf("[%relay.") >= 0,
+    text.indexOf("[%relay.") >= 0 || text.indexOf("%relay.") >= 0,
```
[the remark this answers is not in the document]
Comment (sorenhoyer) [reply packages/reason-relay/language-plugin/src/FindGraphQLTags.ts]: Of course. I've changed it. :-)

**Comment by zth** [comment]
@sorenhoyer one final thing 😅  Could you add an entry to the changelog? Please tag and thank yourself for the work in there too! ;)

**Comment by sorenhoyer** [comment]
The error in the pibeline is probably related to this https://github.com/actions/cache/issues/144#issuecomment-574465667

**Review by zth — APPROVED** [review]
🎉 Awesome, thank you so much for doing this!

**Pull request merged**